src/core/lib/security/credentials/external/external_account_credentials.cc

```diff
@@ -321,7 +335,7 @@ void ExternalAccountCredentials::ExchangeToken(
   memset(&request, 0, sizeof(grpc_http_request));
   grpc_http_header* headers = nullptr;
```

Comment:
Suggest restructuring this a bit to avoid some duplication:

```
const bool add_authorization_header =
    !options_.client_id.empty() && !options_.client_secret.empty();
request.hdr_count = add_authorization_header ? 3 : 2;
auto* headers = static_cast<grpc_http_header*>(
    gpr_malloc(sizeof(grpc_http_header) * request.hdr_count));
headers[0].key = gpr_strdup("Content-Type");
headers[0].value = gpr_strdup("application/x-www-form-urlencoded");
headers[1].key = gpr_strdup("x-goog-api-client");
headers[1].value = gpr_strdup(MetricsHeaderValue().c_str());
if (add_authorization_header) {
  std::string raw_cred =
      absl::StrFormat("%s:%s", options_.client_id, options_.client_secret);
  char* encoded_cred =
      grpc_base64_encode(raw_cred.c_str(), raw_cred.length(), 0, 0);
  std::string str = absl::StrFormat("Basic %s", std::string(encoded_cred));
  headers[2].key = gpr_strdup("Authorization");
  headers[2].value = gpr_strdup(str.c_str());
}
```